### PR TITLE
[WIP] Remove generation of jks files for logging from ansible

### PR DESCRIPTION
**Pull request by ewolinetz**
This will have to follow a change to the logging es images so that they can read in these files (if they exist) and generate the jks files upon start up

cc @jcantrill @richm

### CODE REVIEW

File: roles/openshift_logging/tasks/generate_pems.yaml [hunk 1]
```diff
@@ -9,23 +9,12 @@
   register: cert_file
   check_mode: no
 
-- name: Creating cert req for {{component}}
```
Comment (ewolinetz) [inline roles/openshift_logging/tasks/generate_pems.yaml]: this was left over code from trying to generate pkcs12 in ansible for es + sg

File: roles/openshift_logging/tasks/generate_certs.yaml [hunk 2]
```diff
@@ -134,8 +134,35 @@
...
+        - "logging-es.{{openshift_logging_namespace}}.svc.cluster.local"
+        - "logging-es-ops"
+        - "logging-es-cluster-ops"
+        - "logging-es-ops.{{openshift_logging_namespace}}.svc.cluster.local"
+        - "{{ openshift_logging_es_hostname | default('') }}"
+        - "{{ openshift_logging_es_ops_hostname | default('') }}"
+  loop_control:
+    loop_var: node_name
+
+- debug: var={{ blah.blah }}
```
Comment (ewolinetz) [inline roles/openshift_logging/tasks/generate_certs.yaml]: will remove when ready to merge

**Comment by richm** [comment]
what about this stuff: https://github.com/openshift/openshift-ansible/blob/master/roles/openshift_logging/tasks/generate_jks.yaml#L7

Is the subjectAltName format limitation only a problem with jks?

File: roles/openshift_logging/tasks/generate_es_pems.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+- name: Checking for {{ component }}.crt
+  stat: path="{{ generated_certs_dir }}/{{ component }}.crt"
+  register: cert_file
+  check_mode: no
+
+- set_fact:
+    cert_ext: "{{ certs | strflist(pattern='DNS.{@}:{}', index_start=2, ignore_empty_key=True) }}"
+
+- set_fact:
+    cert_ext: "{{ cert_ext + ['RID.1:1.2.3.4.5.5'] }}"
```
Comment (jcantrill) [inline roles/openshift_logging/tasks/generate_es_pems.yaml]: Do we know where this is configured in SG.  This PR[1] was the first attempt to use the OID specific to our organization

[1] https://github.com/openshift/origin-aggregated-logging/pull/319

**Comment by ewolinetz** [comment]
@richm I'm not sure, we will have to test that to verify. I'm not sure if it was just an issue when creating a key from scratch or if it also impacts when we import a key...

File: roles/openshift_logging/tasks/generate_es_pems.yaml (hunk 3 above)
Comment (ewolinetz) [reply roles/openshift_logging/tasks/generate_es_pems.yaml]: https://github.com/floragunncom/search-guard/blob/fd43977b779081555ab4d1b5cc775f5e2f34349a/src/main/java/com/floragunn/searchguard/support/ConfigConstants.java#L155

Comment (ewolinetz) [reply roles/openshift_logging/tasks/generate_es_pems.yaml]: Though, if we aren't going to auto gen our certs each time, i'd be a little hesitant to set that... yet.

**Comment by ewolinetz** [comment]
logging tests will fail until https://github.com/openshift/origin-aggregated-logging/pull/1252 is merged in

**Comment by ewolinetz** [comment]
es image is missing openssl..

**Comment by ewolinetz** [comment]
fyi @t0ffel the operator may need to do this if we cannot leverage the cert signing service to do this for us

**Comment by vrutkovs** [comment]
Master branch is closed! A major refactor is ongoing in devel-40. Changes for 3.x should be made directly to the latest release branch they're relevant to and backported from there.

**Pull request closed without merging**